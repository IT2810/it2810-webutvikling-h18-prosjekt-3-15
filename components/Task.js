import React, { Component } from 'react';
import {
    View,
    Text,
    StyleSheet,
    TouchableOpacity,
    AsyncStorage,
} from 'react-native';
import { CheckBox } from 'react-native-elements';

// alle disse kommentarene er hvordan jeg tror det kan løses ish, men vet ikke hvor jeg skal starte
// eller hvordan jeg skal implementere det (:

//const checkedSave = { 'checked' : 'false'};

//const checkToBeSaved = { 'checked' : this.state.checked};

//const existingCheck = await AsyncStorage.getItem('checkedbox');

//let newChecked = JSON.parse(existingCheck);
//if( !newChecked ){
//    newChecked = []
//}

//newChecked.push( checkToBeSaved )

//await AsyncStorage.setItem('checkedbox', JSON.stringify(newChecked) )
//    .then( () => { console.log ('Successfully saved')} )
//    .catch( () => { console.log('There was an error saving the CheckedState.')})

export default class Task extends Component {
    constructor(props){
        super(props);
    }

    // disse saver bare strings ?? hvordan booleans da ? lagre hvem som er checked.
    // parse til JSON også tilbake ? how.
    /*
    async getChecked(){
        try{
            let value = await AsyncStorage.getItem('checkedbox', value);
            console.log('value : ' + value);
            let parseValue = JSON.parse(value);
            console.log('parsedValue : ' + parseValue);
            this.setState({checked: parseValue})

        } catch (error){
            console.log("Error retrieving data " + error);
        }
    }

    async saveChecked(value){
        try{
            await AsyncStorage.setItem('checkedbox', JSON.stringify(this.state.checked));
            console.log("save value : " + value);
        }catch(error){
            console.log("Error saving data " + error);
        }
    } */

    checkboxClicked(){
        /*this.setState({checked: !this.state.checked});
        this.saveChecked(this.state.checked);
        console.log(this.state.checked); */
        this.props.callback(this.props.taskKey);
    }
/*
    componentDidMount() {
        this.getChecked()
    } */

    render() {
        return (
            <View key={this.props.keyval} style={styles.task}>
                <Text style={styles.taskText}>{this.props.date}</Text>
                <Text style={styles.taskText}>{this.props.task}</Text>
                <View style={styles.checkbox}>
                <CheckBox style={styles.checkbox} checked={this.props.checked}
                          onPress={() => this.checkboxClicked()} />
                </View>
                <TouchableOpacity onPress={this.props.deleteMethod} style={styles.taskDelete}>
                    <Text style={styles.taskDeleteText}>Delete</Text>
                </TouchableOpacity>
            </View>
        );
    }
}

const styles = StyleSheet.create({
    task: {
        position: 'relative',
        padding: 20,
        paddingRight: 100,
        borderBottomWidth:2,
        borderBottomColor: '#ededed'
    },
    taskText: {
        paddingLeft: 20,
        borderLeftWidth: 15,
        paddingRight: 5,
        borderLeftColor: '#4286f4',
        opacity: 0.7,
    },
    taskDelete: {
        position: 'absolute',
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: '#f44141',
        padding: 10,
        top: 15,
        bottom: 15,
        right: 10
    },
    taskDeleteText: {
        color: 'white'
    },
    checkbox: {
        position: 'absolute',
        justifyContent: 'center',
        alignItems: 'center',
        top: 10,
        bottom: 10,
        right: 65
    }
});
