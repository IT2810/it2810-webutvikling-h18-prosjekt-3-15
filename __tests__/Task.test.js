import React from 'react';
import Task from '../components/Task.js';
import renderer from 'react-test-renderer';

import ShallowRenderer from 'react-test-renderer/shallow';

it('renders correctly', () => {
    const renderer = new ShallowRenderer();
    const tree = renderer.render(<Task/>);
    expect(tree).toMatchSnapshot();
});
