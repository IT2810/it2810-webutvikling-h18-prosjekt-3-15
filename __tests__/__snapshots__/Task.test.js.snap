// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "borderBottomColor": "#ededed",
      "borderBottomWidth": 2,
      "padding": 20,
      "paddingRight": 100,
      "position": "relative",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "borderLeftColor": "#4286f4",
        "borderLeftWidth": 15,
        "opacity": 0.7,
        "paddingLeft": 20,
        "paddingRight": 5,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "borderLeftColor": "#4286f4",
        "borderLeftWidth": 15,
        "opacity": 0.7,
        "paddingLeft": 20,
        "paddingRight": 5,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "bottom": 10,
        "justifyContent": "center",
        "position": "absolute",
        "right": 65,
        "top": 10,
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#fafafa",
          "borderColor": "#ededed",
          "borderRadius": 3,
          "borderWidth": 1,
          "margin": 5,
          "marginLeft": 10,
          "marginRight": 10,
          "opacity": 1,
          "padding": 10,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            },
            false,
            false,
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={false}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "color": "#bfbfbf",
                "fontSize": 24,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          ï‚–
        </Text>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {},
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "color": "#43484d",
                  "fontWeight": "bold",
                  "marginLeft": 10,
                  "marginRight": 10,
                },
                undefined,
                undefined,
              ],
            ]
          }
        />
      </View>
    </View>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#f44141",
        "bottom": 15,
        "justifyContent": "center",
        "opacity": 1,
        "padding": 10,
        "position": "absolute",
        "right": 10,
        "top": 15,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
        }
      }
    >
      Delete
    </Text>
  </View>
</View>
`;
